evm_arithmetization/src/generation/state.rs

```diff
@@ -196,6 +196,13 @@ pub(crate) trait State<F: RichField> {
             let registers = self.get_registers();
             let pc = registers.program_counter;
 
+            // If we reached init_access_lists, we need to clear the pointers
+            let init_access_lists = KERNEL.global_labels["init_access_lists"];
+            if running && pc == init_access_lists  && registers.is_kernel {
+                let gen_state = self.get_mut_generation_state();
+                gen_state.access_lists_ptrs = LinkedListPtrs::default();
+            }
```

Comment:
So I don't know how _heavy_ this additional check is, but given this is run on literally every operation, while reinitialization are extremely rare in comparison, couldn't we just add a `PROVER_INPUT(access_list::reset)` POP` line in the associated macro in the kernel, and have this prover input do the flushing for us? I don't think we'd suffer from 2 additional cycles per txn.

Reply:
Yep, I think that's a better solution. Thanks!